tests/test_jointcal.py

```diff
@@ -71,10 +78,10 @@ def test_iterateFit_badFinalChi2(self):
         self.jointcal.log = log
         self.fitter.computeChi2.return_value = self.badChi2
 
-        chi2 = self.jointcal._iterate_fit(self.associations, self.fitter,
+        chi2 = self.jointcal._iterate_fit(self.associations, self.fitter, self.model,
                                           self.maxSteps, self.name, self.whatToFit)
         self.assertEqual(chi2, self.badChi2)
-        log.info.assert_called_with("Fit completed with: %s", str(self.badChi2))
+        log.info.assert_called_with("%s %s", "Fit completed", str(self.badChi2))
```

Comment:
Given that logging is something that happens on the side, should you really be testing for it?

Reply:
Do we really assert that "logging happens on the side" in the stack? It's the only way to output some information (like whether something completed successfully, in this case).

Reply:
The assumption behind logging (and frameworks like log4* specifically) is that the program is running with little to no human oversight, possibly continuously, and that logging statements are only there to provide information that might be useful for reviewing things after the fact (especially if something went wrong). In other words, it should be possible to remove all logging statements and the program will still be correct.

Logging can be misused as a replacement for actual program output, but I don't think that's something we should encourage. That said, I don't know of any Stack programs that are _supposed_ to produce text output -- we're very much operating in the non-interactive, machine-readable regime.